solidity/test/l2/L2TBTC.test.ts

```diff
@@ -0,0 +1,261 @@
+import { SignerWithAddress } from "@nomiclabs/hardhat-ethers/signers"
+import { randomBytes } from "crypto"
+import { ethers, getUnnamedAccounts, helpers, waffle } from "hardhat"
+import { expect } from "chai"
+import { ContractTransaction } from "ethers"
+
+import type { L2TBTC } from "../../typechain"
+
+const { createSnapshot, restoreSnapshot } = helpers.snapshot
+
+describe("L2TBTC", () => {
+  const fixture = async () => {
+    const { deployer, governance } = await helpers.signers.getNamedSigners()
+
+    const accounts = await getUnnamedAccounts()
+    const minter = await ethers.getSigner(accounts[1])
+    const thirdParty = await ethers.getSigner(accounts[2])
+
+    const deployment = await helpers.upgrades.deployProxy(
+      // Hacky workaround allowing to deploy proxy contract any number of times
+      // without clearing `deployments/hardhat` directory.
+      // See: https://github.com/keep-network/hardhat-helpers/issues/38
+      `L2TBTC_${randomBytes(8).toString("hex")}`,
```

Comment:
Not super proud of it but I had to move forward. May be replaced with a flag if we do https://github.com/keep-network/hardhat-helpers/issues/38.